vector/src/main/java/im/vector/util/RoomUtils.java

```diff
@@ -663,9 +666,81 @@ public boolean onMenuItemClick(final MenuItem item) {
      * Display a confirmation dialog when user wants to leave a room
      *
      * @param context
+     * @param session
+     * @param room
      * @param onClickListener
      */
-    public static void showLeaveRoomDialog(final Context context, final DialogInterface.OnClickListener onClickListener) {
+    public static void showLeaveRoomDialog(final Context context, @NonNull final MXSession session, @NonNull final Room room, @NonNull final DialogInterface.OnClickListener onClickListener) {
+        // Check whether the user is the last admin.
+        PowerLevels powerLevels = room.getState().getPowerLevels();
+        if (null != powerLevels) {
+            int powerLevel = powerLevels.getUserPowerLevel(session.getMyUserId());
+            boolean isAdmin = (powerLevel >= CommonActivityUtils.UTILS_POWER_LEVEL_ADMIN);
+
+            if (isAdmin) {
+                // Check other members's power level.
+                room.getJoinedMembersAsync(new ApiCallback<List<RoomMember>>() {
+
+                    private void onDone(boolean isLastAdmin) {
+                        String message;
+                        if (isLastAdmin) {
+                            message = context.getString(R.string.tchap_room_admin_leave_prompt_msg);
+                        } else {
+                            message = context.getString(R.string.room_participants_leave_prompt_msg);
+                        }
+
+                        new AlertDialog.Builder(context)
+                                .setTitle(R.string.room_participants_leave_prompt_title)
+                                .setMessage(message)
+                                .setPositiveButton(R.string.leave, new DialogInterface.OnClickListener() {
+                                    @Override
+                                    public void onClick(DialogInterface dialog, int which) {
+                                        onClickListener.onClick(dialog, which);
```

Comment:
no need to wrap in another DialogInterface.OnClickListener(), you can use directly `onClickListener`